### PR TITLE
chore: configure renovate

### LINKED ISSUE #1398: Use an auto dependency updater
**Is your feature request related to a problem? Please describe.**

Dependency updates are currently a manual process.
With js packages the process is run `npm outdated` and copy the new versions into *package.json*
With gradle the process is lookup each dependency on maven central to see if a new version has been published.

**Describe the solution you'd like**

On a regular basis, have pull requests automatically opened with new version numbers.

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

Configure renovate to open pull requests for Java and JavaScript dependency updates.
Renovate will open a new pull request for each version update.

The output will look something like
screenshot_2018-11-05 christianmurphy uportal

resolves #1398

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": [
+    "config:base",
+    ":rebaseStalePrs",
+    ":preserveSemverRanges",
```
Comment (ChristianMurphy) [inline renovate.json]: This asks renovate to not modify any of the version ranges that have been set.

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    ":semanticPrefixChore",
```
Comment (ChristianMurphy) [inline renovate.json]: this prefixes each PR commit message with `chore(deps):`

File: renovate.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    ":disableRateLimiting"
```
Comment (ChristianMurphy) [inline renovate.json]: This keeps allows renovate to open as many update PRs as needed. (will be a lot at first)

File: renovate.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  ],
+  "gradle": {
+    "enabled": true
+  },
+  "renovateFork": true
```
Comment (ChristianMurphy) [inline renovate.json]: This allows renovate to be run on forks of uPortal

File: renovate.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": [
+    "config:base",
+    ":rebaseStalePrs",
+    ":preserveSemverRanges",
+    ":semanticPrefixChore",
+    ":disableRateLimiting"
+  ],
+  "gradle": {
+    "enabled": true
```
Comment (ChristianMurphy) [inline renovate.json]: This enables gradle support, currently in beta.

File: renovate.json (hunk 4 above)
Comment (jonathanmtran) [reply renovate.json]: Why would we want Renovate to look at forks? I'm not seeing the value of this ...

Comment (ChristianMurphy) [reply renovate.json]: No worries, it's a confusing option.

**Pull request merged**